libjob_queue/src/lsf_driver.c

```diff
@@ -330,6 +330,90 @@ static void lsf_driver_assert_submit_method( const lsf_driver_type * driver ) {
 
 
 
+/*
+  The resource request string contains spaces, and when passed
+  through the shell it must be protected with \"..\"; this applies
+  when submitting to a remote lsf server with ssh. However when
+  submitting to the local workstation using a bsub command the
+  command will be invoked with the util_spawn() command - and no
+  shell is involved. In this latter case we must avoid the \"...\"
+  quoting.
+*/
+static char * construct_quoted_resource_string(lsf_driver_type * driver) {
+  stringlist_type * select_list = stringlist_alloc_new();
+  if (stringlist_get_size(driver->exclude_hosts) > 0) {
+    for (int i = 0; i < stringlist_get_size(driver->exclude_hosts); i++) {
+      char * exclude_host = util_alloc_sprintf("hname!='%s'",
+                                               stringlist_iget(driver->exclude_hosts, i));
+      stringlist_append_owned_ref(select_list, exclude_host);
+    }
+  }
```

Comment:
Apart from the `if` test on line 354, which could be based on the `driver->exclude_hosts` variable, we only need the `select_list` in the form of the ` && ` joined string; I suggest to extract this to a separate function:

```C
char* lsf_driver_alloc_select_string( const lsf_driver_type * driver) {

}
```
which would return the string: `"hname!='host1' && hname!='host2' && hname!='host3'"`